libs/sql-parser/src/main/java/io/crate/sql/tree/FunctionCall.java

```diff
@@ -32,21 +32,32 @@
     private final List<Expression> arguments;
     private final Optional<Window> window;
     private final Optional<Expression> filter;
+    private final Boolean ignoreNulls;
```

Comment:
Could this be normalized to a `boolean` with only 2 states (true or false) instead of 3 (incl. null) by implying a default value - or is `null` a distinct case that we need to handle differently?

Reply:
null should imply that the user did not provide any flag at all. For example, if ignore null is set for `rank()` I should throw.